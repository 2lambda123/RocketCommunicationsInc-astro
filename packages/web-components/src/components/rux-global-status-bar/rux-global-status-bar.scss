@use '../../global/tokens/colors-dark' as variables;

:host {
    display: block;
    position: sticky;
    z-index: 50;
    top: 0;
    left: 0;
    height: var(--spacing-24);
    width: -webkit-fill-available;
    width: -moz-available;
    width: stretch;
    padding: 0 var(--spacing-6);
    box-sizing: border-box;
    background-color: var(--gsb-color-background);
    -webkit-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    user-select: none;
    contain: layout; /* This improves CSS performance see: https://developers.google.com/web/updates/2016/06/css-containment */
}

:host([hidden]) {
    display: none;
}

header {
    overflow: hidden;
    display: flex;
    height: 100%;
    width: 100%;
    align-items: center;
    justify-content: space-between;
    box-sizing: border-box;
}

slot[name='left-side'] > *,
::slotted(*[slot='left-side']) {
    margin-right: var(--spacing-2);
}

slot[name='left-side'] > rux-icon,
::slotted(rux-icon[slot='left-side']) {
    height: calc(var(--spacing-8) + var(--spacing-050)); //38px
}

slot[name='left-side'] > .shifted-up,
::slotted(.shifted-up[slot='left-side']) {
    height: calc(var(--spacing-14) + var(--spacing-050)); //58px
}

slot[name='right-side'] > rux-monitoring-icon,
::slotted(rux-monitoring-icon[slot='right-side']) {
    display: flex;
    align-items: center;
    padding-right: calc(var(--spacing-4) + var(--spacing-2)); //24px
    padding-top: var(--spacing-2);
}

.app-meta,
::slotted(*[slot='app-meta']) {
    display: inline-flex;
    flex-wrap: wrap;
    padding: var(--spacing-0);
    margin: var(--spacing-0) auto var(--spacing-0) var(--spacing-0);
    color: var(--color-palette-neutral-000);
    white-space: nowrap;
}

.app-info-wrapper {
    flex-basis: 100%;
    display: flex;
    align-items: baseline;
    min-height: var(--spacing-10); //40px
    padding-bottom: var(--spacing-1);
    min-width: 170px;
}

.app-meta h1 {
    margin-bottom: var(--spacing-0);
    margin-top: var(--spacing-0);
    font-family: var(--font-heading-1-font-family);
    font-size: var(--font-heading-1-font-size);
    font-weight: var(--font-heading-1-font-weight);
    letter-spacing: var(--font-heading-1-letter-spacing);
    line-height: var(--line-height-2xl);
}

.app-meta h1.app-domain {
    margin-right: var(--spacing-2);
    font-family: var(--font-heading-1-bold-font-family);
    font-size: var(--font-heading-1-bold-font-size);
    font-weight: var(--font-heading-1-bold-font-weight);
    letter-spacing: var(--font-heading-1-bold-letter-spacing);
    line-height: var(--line-height-2xl);
}

.app-name {
    margin-right: var(--spacing-2); //8px
}

.app-version {
    display: inline-block;
    font-family: var(--font-heading-6-font-family);
    font-size: var(--font-heading-6-font-size);
    font-weight: var(--font-heading-6-font-weight);
    letter-spacing: var(--font-heading-6-letter-spacing);
    line-height: var(--font-heading-6-line-height);
}

.app-state-wrapper {
    display: flex;
}

// all temp classes and their subsequent styling will be replaced with internal component styles.
.app-state {
    box-sizing: border-box;
    color: var(--color-palette-neutral-000);
    background-color: var(--color-palette-neutral-1000);
    border-radius: var(--radius-base);
    font-family: var(--font-body-2-font-family);
    font-size: var(--font-body-2-font-size);
    font-weight: var(--font-body-2-font-weight);
    letter-spacing: var(--font-body-2-letter-spacing);
    // padding: var(--spacing-050) calc(var(--spacing-1) + var(--spacing-050)); //2px 6px
    // padding: 3px 8px;
    padding: calc(var(--spacing-050) + var(--spacing-025)) var(--spacing-2); //3px 8px
    text-align: center;
    // height: 22px;
    margin-right: var(--spacing-2); //6px
    white-space: nowrap;
}

.username {
    box-sizing: border-box;
    flex-basis: 50%;
    font-family: var(--font-body-2-font-family);
    font-size: var(--font-body-2-font-size);
    font-weight: var(--font-body-2-font-weight);
    letter-spacing: var(--font-body-2-letter-spacing);
    padding: calc(var(--spacing-050) + var(--spacing-025)) var(--spacing-0); //3px 0
    height: calc(var(--spacing-6) - var(--spacing-050)); //22px
}

.slotted-content {
    display: inline-flex;
    justify-content: center;
    flex-basis: 100%;
}

//Preserves dark theme defaults for slotted elements in GSB since content must always be rendered in dark theme.
::slotted(*) {
    @include variables.root-variables;
}
