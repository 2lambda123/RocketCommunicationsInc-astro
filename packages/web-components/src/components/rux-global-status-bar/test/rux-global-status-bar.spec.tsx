import { newSpecPage } from '@stencil/core/testing'
import { RuxGlobalStatusBar } from '../rux-global-status-bar'

describe('rux-global-status-bar', () => {
    it('renders', async () => {
        const page = await newSpecPage({
            components: [RuxGlobalStatusBar],
            html: `<rux-global-status-bar></rux-global-status-bar>`,
        })
        expect(page.root).toMatchSnapshot()
    })

    it('renders with icon and app meta', async () => {
        const page = await newSpecPage({
            components: [RuxGlobalStatusBar],
            html: `<rux-global-status-bar include-icon app-state="App State" username="Username" app-domain="ASTRO" app-name="Test App Name" app-version="test v1.0"></rux-global-status-bar>`,
        })
        expect(page.root).toMatchSnapshot()
    })

    it('renders with icon, app meta and slotted content', async () => {
        const page = await newSpecPage({
            components: [RuxGlobalStatusBar],
            html: `<rux-global-status-bar include-icon app-state="App State" username="Username" app-domain="ASTRO" app-name="Test App Name" app-version="test v1.0">
                <div>Tab links<div>
                <button slot="right-side">Emergency shut off</button>
            </rux-global-status-bar>`,
        })
        page.waitForChanges()
        expect(page.root).toMatchSnapshot()
    })
})
