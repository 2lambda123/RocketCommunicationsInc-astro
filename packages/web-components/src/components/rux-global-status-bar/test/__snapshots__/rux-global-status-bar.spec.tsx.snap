// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rux-global-status-bar renders 1`] = `
<rux-global-status-bar menu-icon="apps">
  <mock:shadow-root>
    <header part="container">
      <slot name="left-side"></slot>
      <slot name="app-meta"></slot>
      <div class="slotted-content">
        <slot></slot>
      </div>
      <slot name="right-side"></slot>
    </header>
  </mock:shadow-root>
</rux-global-status-bar>
`;

exports[`rux-global-status-bar renders with icon and app meta 1`] = `
<rux-global-status-bar app-domain="ASTRO" app-name="Test App Name" app-state="App State" app-version="test v1.0" include-icon="" menu-icon="apps" username="Username">
  <mock:shadow-root>
    <header part="container">
      <slot name="left-side">
        <rux-icon class="shifted-up" icon="apps" size="small"></rux-icon>
      </slot>
      <slot name="app-meta">
        <div class="app-meta">
          <div class="app-info-wrapper">
            <h1 class="app-domain">
              ASTRO
            </h1>
            <h1 class="app-name">
              TEST APP NAME
            </h1>
            <span class="app-version">
              test v1.0
            </span>
          </div>
          <div class="app-state-wrapper">
            <div class="app-state" part="app-state" style="background-color: var(--color-global-tag-tag1-600);">
              App State
            </div>
            <div class="username" part="username">
              Username
            </div>
          </div>
        </div>
      </slot>
      <div class="slotted-content">
        <slot></slot>
      </div>
      <slot name="right-side"></slot>
    </header>
  </mock:shadow-root>
</rux-global-status-bar>
`;

exports[`rux-global-status-bar renders with icon, app meta and slotted content 1`] = `
<rux-global-status-bar app-domain="ASTRO" app-name="Test App Name" app-state="App State" app-version="test v1.0" include-icon="" menu-icon="apps" username="Username">
  <mock:shadow-root>
    <header part="container">
      <slot name="left-side">
        <rux-icon class="shifted-up" icon="apps" size="small"></rux-icon>
      </slot>
      <slot name="app-meta">
        <div class="app-meta">
          <div class="app-info-wrapper">
            <h1 class="app-domain">
              ASTRO
            </h1>
            <h1 class="app-name">
              TEST APP NAME
            </h1>
            <span class="app-version">
              test v1.0
            </span>
          </div>
          <div class="app-state-wrapper">
            <div class="app-state" part="app-state" style="background-color: var(--color-global-tag-tag1-600);">
              App State
            </div>
            <div class="username" part="username">
              Username
            </div>
          </div>
        </div>
      </slot>
      <div class="slotted-content">
        <slot></slot>
      </div>
      <slot name="right-side"></slot>
    </header>
  </mock:shadow-root>
  <div>
    Tab links
    <div>
      <button slot="right-side">
        Emergency shut off
      </button>
    </div>
  </div>
</rux-global-status-bar>
`;
